### ISSUE TITLE
creating a :ns-group that matches a namespace prevents using :config-in-ns 

### ISSUE THREAD

**Issue by leafgarland** [clj-kondo/clj-kondo#1737]
**version**

clj-kondo v2022.06.22

**platform**

homebrew M1 install

**problem**

When a `:ns-group` is defined, any `:config-in-ns` settings for namespaces that match the group are ignored 

**repro**

This test shows that `:config-in-ns` setting turns off the unresolved-symbol linter
```sh
# command
echo '(ns my.namespace) x y z' | clj-kondo --lint - --config '{:config-in-ns {my.namespace {:linters {:unresolved-symbol {:level :off}}}}}'

# output
linting took 8ms, errors: 0, warnings: 0
```

If we run the same config but with a :ns-group defined then the settings in `:config-in-ns` are ignored.

```sh
# command
echo '(ns my.namespace) x y z' | clj-kondo --lint - --config '{:ns-groups [{:pattern "my.*" :name mine}] :config-in-ns {my.namespace {:linters {:unresolved-symbol {:level :off}}}}}'

# output
<stdin>:1:19: error: Unresolved symbol: x
<stdin>:1:21: error: Unresolved symbol: y
<stdin>:1:23: error: Unresolved symbol: z
linting took 8ms, errors: 3, warnings: 0
```

**expected behavior**

I would like to be able to define some settings for whole groups and some settings for individual namespaces (even if they are in a group).

Thanks!

**Comment by borkdude** [comment]
Makes sense!
